src/scripts/main.js

```diff
@@ -1,11 +1,30 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handleEvent = () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+      element.removeEventListener(eventName, handleEvent);
+    };
+
+    element.addEventListener(eventName, handleEvent);
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const messageContainer = document.createElement('div');
+
+  messageContainer.classList.add('message-container');
+  document.body.appendChild(messageContainer);
+
+  const messageEl = document.createElement('div');
+
+  messageEl.classList.add('message');
+  messageEl.textContent = message;
+
+  messageContainer.appendChild(messageEl);
 };
```

Comment:
The 'printMessage' function creates a new 'message-container' div every time it is called. This could result in many divs being created if the function is called multiple times. Consider creating the 'message-container' div once outside of the function and only creating and appending new 'message' divs within the function.